index.js

```diff
@@ -689,6 +696,47 @@ class Hyperbee extends ReadyResource {
     return (this._checkout <= this.core.length || this._checkout <= 1) ? this.core.snapshot() : this.core.session({ snapshot: false })
   }
 
+  async clearUnlinked (options = {}) {
+    await this.ready()
+
+    const { gte = 0, lt = this.version - 1 } = options
+    const checkout = this.version
+
+    const prev = this.batch({
+      wait: false,
+      checkout: gte
+    })
+
+    const b = this.batch({
+      wait: false,
+      checkout
+    })
+
+    const ite = new LocalBlockIterator(b, { gte, lt })
+    await ite.open()
+
+    while (true) {
+      const data = await ite.next()
+      if (!data) break
+
+      if (!(await isLinked(b, data))) {
+        await this.core.clear(data.seq)
+      }
+
+      const prevNode = await prev.get(data.key, { finalize: false }).catch(toNull)
```

Comment:
`finalize` is an optimisation to avoid decoding the node, right, because we want to access the hypercore data like the seq? (if so all good, if not I misunderstood this part of the code)